lib/src/main/java/com/squareup/cycler/DataSource.kt

```diff
@@ -1,26 +1,28 @@
 package com.squareup.cycler
 
 /**
- * Very minimal interface for the item list.
- * You can use extensions [Array.toDataSource] and [List.toDataSource].
+ * If you happened to have a custom implementation of [DataSource] you can
+ * subclass [AbstractList] which has the same abstract API surface.
+ *
+ * Ex:
+ * fun <T> Array<T>.toDataSource(): DataSource<T> {
+ *   return object : AbstractList<T>() {
+ *     override fun get(i: Int): T = this@toDataSource[i]
+ *     override val size get() = this@toDataSource.size
+ *   }
+ * }
  */
-interface DataSource<out T> {
-  operator fun get(i: Int): T
-  val size: Int
-  val isEmpty: Boolean
-    get() = size == 0
-}
+@Deprecated(message = "DataSource is now just an alias for List and will be removed in the future.")
+typealias DataSource<T> = List<T>
 
-fun <T> List<T>.toDataSource(): DataSource<T> {
-  return object : DataSource<T> {
-    override fun get(i: Int): T = this@toDataSource[i]
-    override val size get() = this@toDataSource.size
-  }
-}
+@Deprecated(
+  message = "Converting to DataSource is no longer necessary.",
+  replaceWith = ReplaceWith("this")
+)
+fun <T> List<T>.toDataSource(): List<T> = this
 
-fun <T> Array<T>.toDataSource(): DataSource<T> {
-  return object : DataSource<T> {
-    override fun get(i: Int): T = this@toDataSource[i]
-    override val size get() = this@toDataSource.size
-  }
-}
+@Deprecated(
+  message = "Converting to DataSource is no longer necessary.",
+  replaceWith = ReplaceWith("toList()")
```

Comment:
Probably `this.`toList() is necessary.

Reply:
Should it be `asList()`?

Reply:
> Should it be `asList()`?

Good question. 

It looks to me like they both do similar things when size >  1.

Reply:
https://stackoverflow.com/questions/48411392/how-are-kotlin-arrays-tolist-and-aslist-different
asList: uses the original array as source.
toList: copies the data.
It really depends on what you use, but the original implementation of this toDatasource was using the original list as source, so asList sounds appropriate.

Reply:
Turns out `asList` maintains a reference to the original array whereas `toList` makes a copy of the original array. 

Thus when using `asList`, any changes made to the backing `Array` will be reflected in the new list. 

I think `toList` is probably what we want here.

Reply:
> Probably `this.`toList() is necessary.

I double checked and the `this` is not necessary.

Reply:
@zach-klippenstein @helios175 The original implementation was essentially an `asList`. As such I'm changing this accordingly to avoid any unintended consequences.